pkg/cloudprovider/cloudprovider.go

```diff
@@ -201,6 +226,37 @@ func kubeDNSIP(ctx context.Context, kubernetesInterface kubernetes.Interface) (n
 	return kubeDNSIP, nil
 }
 
+func (c *CloudProvider) isAMIDrifted(ctx context.Context, machine *corev1alpha1.Machine, provisioner *v1alpha5.Provisioner, nodeTemplate *v1alpha1.AWSNodeTemplate) (bool, error) {
+	instanceTypes, err := c.GetInstanceTypes(ctx, provisioner)
+	if err != nil {
+		return false, fmt.Errorf("getting instanceTypes, %w", err)
+	}
+	nodeInstanceType, found := lo.Find(instanceTypes, func(instType *cloudprovider.InstanceType) bool {
+		return instType.Name == machine.Labels[v1.LabelInstanceTypeStable]
+	})
+	if !found {
+		return false, fmt.Errorf("getting node instance type")
+	}
+	if nodeTemplate.Spec.LaunchTemplateName != nil {
+		return false, nil
+	}
+	amis, err := c.amiProvider.Get(ctx, nodeTemplate, []*cloudprovider.InstanceType{nodeInstanceType},
+		amifamily.GetAMIFamily(nodeTemplate.Spec.AMIFamily, &amifamily.Options{}))
+	if err != nil {
+		return false, fmt.Errorf("getting amis, %w", err)
+	}
+	// Get InstanceID to fetch from EC2
+	instanceID, err := utils.ParseMachineInstanceID(machine)
+	if err != nil {
+		return false, err
+	}
+	instance, err := c.instanceProvider.getInstance(ctx, *instanceID)
+	if err != nil {
+		return false, fmt.Errorf("getting instance, %w", err)
+	}
+	return !lo.Contains(lo.Keys(amis), *instance.ImageId), nil
```

Comment:
wouldn't this say there's no drift if a new version of the AMI is released but is still included in the return of `amiProvider.Get` ? 

i.e.

```
amis, _ := amiProvider.Get(ctx,  nodeTemplate, []*cloudprovider.InstanceType{nodeInstanceType}, amiFamily...) 
-> ["ami-123"]

... time passes ...

amis, _ := amiProvider.Get(ctx,  nodeTemplate, []*cloudprovider.InstanceType{nodeInstanceType}, amiFamily...) 
-> ["ami-234", "ami-123"]
```

In the case above, "ami-123" was first published and then "ami-234" was published, so the amiProvider sorted by most recent, but the ami would be marked as not being drited since "ami-123" is in the list still. 

Should you just check the first element in the amiID's return?